src/file/styles/style/paragraph-style.ts

```diff
@@ -26,40 +17,12 @@ export class StyleForParagraph extends Style {
     private readonly runProperties: RunProperties;
 
     constructor(options: IParagraphStyleOptions) {
-        super({ type: "paragraph", styleId: options.id }, options.name);
+        super({ type: "paragraph", styleId: options.id }, options);
 
         this.paragraphProperties = new ParagraphProperties(options.paragraph);
         this.runProperties = new RunProperties(options.run);
 
         this.root.push(this.paragraphProperties);
         this.root.push(this.runProperties);
-
-        if (options.basedOn) {
```

Comment:
Are these not needed? Why are they deleted?

Reply:
I can see they were moved to the `Style` class

Reply:
most of the shared options between the different types of Style classes were moved to the base class